NullPointerException in GatherSystemInformationTask

Reported as new in 2021.03.28-RC on Google Play:

1 user, Android 8,

```
java.lang.RuntimeException: 
  at android.os.AsyncTask$3.done (AsyncTask.java:365)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at android.os.AsyncTask$SerialExecutor$1.run (AsyncTask.java:257)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1162)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:636)
  at java.lang.Thread.run (Thread.java:784)
Caused by: java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:203)
  at java.util.Arrays$ArrayList.<init> (Arrays.java:3741)
  at java.util.Arrays.asList (Arrays.java:3728)
  at cgeo.geocaching.utils.CollectionStream.of (CollectionStream.java:64)
  at cgeo.geocaching.storage.FileContentAccessor.list (FileContentAccessor.java:59)
  at cgeo.geocaching.storage.ContentStorage.list (ContentStorage.java:213)
  at cgeo.geocaching.storage.ContentStorage.list (ContentStorage.java:203)
  at cgeo.geocaching.storage.FolderUtils.treeWalk (FolderUtils.java:808)
  at cgeo.geocaching.storage.FolderUtils.treeWalk (FolderUtils.java:817)
  at cgeo.geocaching.storage.FolderUtils.treeWalk (FolderUtils.java:804)
  at cgeo.geocaching.storage.FolderUtils.getFolderInfo (FolderUtils.java:162)
  at cgeo.geocaching.utils.SystemInformation.appendPublicFolders (SystemInformation.java:172)
  at cgeo.geocaching.utils.SystemInformation.getSystemInformation (SystemInformation.java:120)
  at cgeo.geocaching.AboutActivity$GatherSystemInformationTask.doInBackground (AboutActivity.java:342)
  at cgeo.geocaching.AboutActivity$GatherSystemInformationTask.doInBackground (AboutActivity.java:316)
  at android.os.AsyncTask$2.call (AsyncTask.java:345)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```

Probably same device, but slightly different stacktrace:

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5041)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5084)
  at android.app.ActivityThread.-wrap20 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2053)
  at android.os.Handler.dispatchMessage (Handler.java:108)
  at android.os.Looper.loop (Looper.java:166)
  at android.app.ActivityThread.main (ActivityThread.java:7529)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:245)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:921)
Caused by: java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:203)
  at java.util.Arrays$ArrayList.<init> (Arrays.java:3741)
  at java.util.Arrays.asList (Arrays.java:3728)
  at cgeo.geocaching.utils.CollectionStream.of (CollectionStream.java:64)
  at cgeo.geocaching.storage.FileContentAccessor.list (FileContentAccessor.java:59)
  at cgeo.geocaching.storage.ContentStorage.list (ContentStorage.java:213)
  at cgeo.geocaching.storage.ContentStorage.list (ContentStorage.java:203)
  at cgeo.geocaching.storage.FolderUtils.treeWalk (FolderUtils.java:808)
  at cgeo.geocaching.storage.FolderUtils.treeWalk (FolderUtils.java:817)
  at cgeo.geocaching.storage.FolderUtils.treeWalk (FolderUtils.java:804)
  at cgeo.geocaching.storage.FolderUtils.getFolderInfo (FolderUtils.java:162)
  at cgeo.geocaching.storage.ContentStorageActivityHelper.getInternationalizedFolderInfoStrings (ContentStorageActivityHelper.java:410)
  at cgeo.geocaching.storage.ContentStorageActivityHelper.continuePersistableFolderSelectionCheckFoldersAreEqual (ContentStorageActivityHelper.java:340)
  at cgeo.geocaching.storage.ContentStorageActivityHelper.handleResultPersistableFolderSelection (ContentStorageActivityHelper.java:323)
  at cgeo.geocaching.storage.ContentStorageActivityHelper.onActivityResult (ContentStorageActivityHelper.java:188)
  at cgeo.geocaching.InstallWizardActivity.onActivityResult (InstallWizardActivity.java:497)
  at android.app.Activity.dispatchActivityResult (Activity.java:7701)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5037)
```

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeHelper.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeHelper.java
@@ -58,7 +58,6 @@ import org.mapsforge.map.rendertheme.XmlRenderThemeStyleLayer;
 import org.mapsforge.map.rendertheme.XmlRenderThemeStyleMenu;
 import org.mapsforge.map.rendertheme.ZipRenderTheme;
 import org.mapsforge.map.rendertheme.ZipXmlThemeResourceProvider;
-import org.xmlpull.v1.XmlPullParserException;
 
 
 /**
@@ -158,7 +157,7 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback, SharedPref
                 ActivityMixin.showApplicationToast(LocalizationUtils.getString(R.string.err_rendertheme_file_unreadable));
                 rendererLayer.setXmlRenderTheme(InternalRenderTheme.OSMARENDER);
                 selectedTheme = null;
-            } catch (final XmlPullParserException e) {
+            } catch (final Exception e) {
                 Log.w("render theme invalid", e);
                 ActivityMixin.showApplicationToast(LocalizationUtils.getString(R.string.err_rendertheme_invalid));
                 rendererLayer.setXmlRenderTheme(InternalRenderTheme.OSMARENDER);

--- a/main/src/cgeo/geocaching/utils/CollectionStream.java
+++ b/main/src/cgeo/geocaching/utils/CollectionStream.java
@@ -61,7 +61,7 @@ public class CollectionStream<T> {
      *
      */
     public static <TT> CollectionStream<TT> of(final TT[] coll) {
-        return new CollectionStream<>(Arrays.asList(coll), false);
+        return new CollectionStream<>(coll == null ? Collections.emptyList() : Arrays.asList(coll), false);
     }
 
     private CollectionStream(final Collection<Object> coll, final boolean forceCopy) {

--- a/tests/src/cgeo/geocaching/utils/CollectionStreamTest.java
+++ b/tests/src/cgeo/geocaching/utils/CollectionStreamTest.java
@@ -27,6 +27,7 @@ public class CollectionStreamTest {
     @Test
     public void testStability() {
         assertThat(CollectionStream.of((Collection<Object>) null).toJoinedString(",")).isEqualTo("");
+        assertThat(CollectionStream.of((Object[]) null).toJoinedString(",")).isEqualTo("");
         assertThat(CollectionStream.of(new Integer[]{1, 2}).map(null).filter(null).toJoinedString(",")).isEqualTo("1,2");
         assertThat(CollectionStream.of(new Integer[]{1, 2}).toJoinedString(null)).isEqualTo("12");
     }

